### PR TITLE
Provide more overloads for the wide string flavour

### PR DESCRIPTION
**Pull request by DanielaE**
Signed-off-by: Daniela Engert <dani@ngrt.de>

### CODE REVIEW

**Comment by DanielaE** [comment]
This code compiles now and is working fine on my machine:
```	auto sResult = fmt::sprintf("Elapsed time: %.2f seconds", 1.23);
	auto wResult = fmt::sprintf(L"Elapsed time: %.2f seconds", 1.23);
	fmt::print(std::cout, "Elapsed time: {:.2f} {}\n", 1.23, "seconds");
	fmt::print(std::wcout, L"Elapsed time: {:.2f} {}\n", 1.23, L"seconds");

	fmt::print("Elapsed time: {:.2f} {}\n", 1.23, "seconds");
	fmt::print_colored(fmt::green, "Elapsed time: {:.2f} {}{}\n", 1.23, 'm', "seconds");
	fmt::print(stdout, "Elapsed time: {:.2f} {}\n", 1.23, "seconds");
	auto n = fmt::printf("Elapsed time: %.2f %cseconds\n", 1.23, 'm');
	n = fmt::fprintf(stdout, "Elapsed time: %.2f %cseconds\n", 1.23, 'm');

	_setmode(_fileno(stdout), _O_U16TEXT); // switch stdout into wchar_t (UTF16) mode

	fmt::print(L"Elapsed time: {:.2f} {}\n", 1.23, L"seconds");
	fmt::print_colored(fmt::green, L"Elapsed time: {:.2f} {}{}\n", 1.23, L'm', L"seconds");
	fmt::print(stdout, L"Elapsed time: {:.2f} {}\n", 1.23, L"seconds");
	n = fmt::printf(L"Elapsed time: %.2f %cseconds\n", 1.23, L'm');
	n = fmt::fprintf(stdout, L"Elapsed time: %.2f %cseconds\n", 1.23, L'm');
```
Should handle issue #721

File: include/fmt/core.h [hunk 1]
```diff
@@ -1093,12 +1093,18 @@ class basic_format_args {
 
 /** An alias to ``basic_format_args<context>``. */
 // It is a separate type rather than a typedef to make symbols readable.
+// It is also required to disambiguate 'vformat_to' overloads
+// in core.h and format.h - which seems a bit brittle!
```
[the remark this answers is not in the document]
Comment (DanielaE) [reply include/fmt/core.h]: Hi Victor,

this is probably no longer true. During my various attempts, to replace
```
template <typename Container>
... vformat_to(std::back_insert_iterator<Container> out,
               string_view format_str, format_args args) {
...
}
``` 
in core.h by 
```
template <typename Container>
... vformat_to(std::back_insert_iterator<Container> out,
      basic_string_view<typename Container::value_type> format_str,
      basic_format_args<typename buffer_context<typename Container::value_type>::type> args) {
...
}
```
to reduce code duplication, I got ambiguous overloads of `vformat_to`in `core.h` and `format.h`. My tests from today show that those more-derived structs don't help me to get along with this problem. Sigh ...

[the remark this answers is not in the document]
Comment (vitaut) [reply include/fmt/core.h]: The problem is not so much due to `basic_format_args`, but due to overloads taking `string_view` in `format.h`. Unfortunately I don't think it's possible to parameterize on character type there: https://godbolt.org/g/bPevXe.

> this is probably no longer true

I've removed the comment then.

**Comment by vitaut** [comment]
> Do you need more test cases? And if so, where?

Thanks a lot for adding tests, I think these are fine for now.

Merged in https://github.com/fmtlib/fmt/commit/2570f1afdc7a81422d216cdc021e1a42d357292c with minor tweaks.

**Pull request closed without merging**